### PR TITLE
fix: Add SNS encryption for FhirWorksAlarm

### PR DESCRIPTION
**Pull request by nguyen102**
Issue #, if available:

Description of changes:
Add SNS encryption for FhirWorksAlarm. As best practices Amazon recommends encrypting SNS topics. 

References
https://aws.amazon.com/blogs/compute/encrypting-messages-published-to-amazon-sns-with-aws-kms/
https://stackoverflow.com/a/58849754
https://aws.amazon.com/premiumsupport/knowledge-center/cloudwatch-receive-sns-for-alarm-trigger/

Checklist:

* [x] Have you successfully deployed to an AWS account with your changes?
* [x] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by nguyen102** [comment]
@Zambonilli , As best practice SNS topics should be encrypted at rest with a KMS key. I wanted to confirm that you're ok with this change, and this wouldn't break anything on your side.

**Review by ssvegaraju — APPROVED** [review]

File: cloudformation/kms.yaml [hunk 1]
```diff
@@ -94,4 +94,31 @@ Resources:
...
+      KeyPolicy:
+        Version: '2012-10-17'
+        Statement:
+          - Sid: 'Enable IAM Root Permissions'
+            Effect: Allow
+            Principal:
+              AWS: !Sub 'arn:${AWS::Partition}:iam::${AWS::AccountId}:root'
+            Action: 'kms:*'
+            Resource: '*'
+          - Sid: 'Allow Cloudwatch to use this Key Policy'
```
Comment (Zambonilli) [inline cloudformation/kms.yaml:116]: we had to have sns.amazonaws.com service principal w/kms:GenerateDataKey* and kms:Decrypt permissions in this policy as well. iirc, that was what allowed sns to encrypt and decrypt when sending to the subscribers.

**Comment by Zambonilli** [comment]
ugh, thanks. I translated our terraform definition for the sns topic here incorrectly and forgot the kms key. This matches what we have in terraform minus the comment about adding sns.amazonaws.com service principal to the key's policy.

**Comment by nguyen102** [comment]
@Zambonilli thanks for double checking. Just to confirm, does this mean the PR as is matches what is currently in your terraform definition?

**Comment by Zambonilli** [comment]
This PR is close to matching our terraform. It's just missing the following Sid in the KMS key's inline policy:
```
- Sid: 'Allow Cloudwatch to use this Key Policy'
            Effect: Allow
            Principal:
              Service: sns.amazonaws.com
            Action:
              - kms:Encrypt
              - kms:Decrypt
              - kms:GenerateDataKey*
            Resource: '*'
```

**Comment by nguyen102** [comment]
Good catch, after further examining the JSON at this [blog post](https://aws.amazon.com/blogs/compute/encrypting-messages-published-to-amazon-sns-with-aws-kms/), I see they include an SID for SNS as well. I've updated the PR.

**Review by carvantes — APPROVED** [review]

**Pull request merged**